src/cmds/docs/cleanup.ts

```diff
@@ -0,0 +1,102 @@
+import type { CommandOptions } from '../../lib/baseCommand';
+
+import chalk from 'chalk';
+import config from 'config';
+
+import Command, { CommandCategories } from '../../lib/baseCommand';
+import createGHA from '../../lib/createGHA';
+import deleteDoc from '../../lib/deleteDoc';
+import getDocs from '../../lib/getDocs';
+import promptTerminal from '../../lib/promptWrapper';
+import readdirRecursive from '../../lib/readdirRecursive';
+import readDoc from '../../lib/readDoc';
+import { getProjectVersion } from '../../lib/versionSelect';
+
+export type Options = {
+  dryRun?: boolean;
+  folder?: string;
+};
+
+function getSlug(filename: string): string {
+  const { slug } = readDoc(filename);
+  return slug;
+}
+
+export default class DocsCleanupCommand extends Command {
+  constructor() {
+    super();
+
+    this.command = 'docs:cleanup';
+    this.usage = 'docs:cleanup <folder> [options]';
+    this.description = 'Delete any docs from ReadMe if their slugs are not found in the target folder.';
+    this.cmdCategory = CommandCategories.DOCS;
+    this.position = 2;
+
+    this.hiddenArgs = ['folder'];
+    this.args = [
+      this.getKeyArg(),
+      this.getVersionArg(),
+      {
+        name: 'folder',
+        type: String,
+        defaultOption: true,
+      },
+      this.getGitHubArg(),
+      {
+        name: 'dryRun',
+        type: Boolean,
+        description: 'Runs the command without creating/updating any docs in ReadMe. Useful for debugging.',
+      },
+    ];
+  }
+
+  async run(opts: CommandOptions<Options>) {
+    await super.run(opts);
+
+    const { dryRun, folder, key, version } = opts;
+
+    if (!folder) {
+      return Promise.reject(new Error(`No folder provided. Usage \`${config.get('cli')} ${this.usage}\`.`));
+    }
+
+    // TODO: should we allow version selection at all here?
+    // Let's revisit this once we re-evaluate our category logic in the API.
+    // Ideally we should ignore this parameter entirely if the category is included.
+    const selectedVersion = await getProjectVersion(version, key);
+
+    Command.debug(`selectedVersion: ${selectedVersion}`);
+
+    // Strip out non-markdown files
+    const files = readdirRecursive(folder).filter(
+      file => file.toLowerCase().endsWith('.md') || file.toLowerCase().endsWith('.markdown')
+    );
+
+    Command.debug(`number of files: ${files.length}`);
+
+    if (!files.length) {
+      const { deleteAll } = await promptTerminal({
+        type: 'confirm',
+        name: 'deleteAll',
+        message: `This action will delete all docs under version ${selectedVersion} from ReadMe, would you like to continue?`,
+      });
+
+      if (!deleteAll) {
+        return Promise.reject(new Error('Aborting, no changes were made.'));
+      }
+    }
+
+    Command.warn(`We're going to delete from ReadMe any document that isn't found in ${folder}.`);
```

Comment:
Should this also have a "are you sure you want to do this?" confirmation prompt? Feels like it might be easy to typo a directory name on this command and wipe all of your docs out without any recourse.

Reply:
Yeah I've sorta felt like we should just wrap this entire command in a confirmation prompt, but I wasn't sure if folks use it in CI. I'll just make that change and if folks request it, we'll create an exception for CI environments.

Reply:
Done: https://github.com/readmeio/rdme/pull/644/commits/ac5580717a7aee33f3efe4034c34f17260c73172